vcpkg-custom-ports/openssl/portfile.cmake

```diff
@@ -0,0 +1,28 @@
+if(EXISTS "${CURRENT_INSTALLED_DIR}/include/openssl/ssl.h")
+    message(FATAL_ERROR "Can't build openssl if libressl/boringssl is installed. Please remove libressl/boringssl, and try install openssl again if you need it.")
+endif()
+
+set(OPENSSL_VERSION 1.1.1n)
+vcpkg_download_distfile(
+    ARCHIVE
+    URLS "https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz" "https://www.openssl.org/source/old/1.1.1/openssl-${OPENSSL_VERSION}.tar.gz"
+    FILENAME "openssl-${OPENSSL_VERSION}.tar.gz"
+    SHA512 1937796736613dcf4105a54e42ecb61f95a1cea74677156f9459aea0f2c95159359e766089632bf364ee6b0d28d661eb9957bce8fecc9d2436378d8d79e8d0a4
```

Comment:
If the openssl version is set to something other than 1.1.1n won't the SHA512 check fail?

Reply:
Yes. The way you deal with this, you set OPENSSL_VERSION=1.1.1q, and set SHA512 to 0. cmake/vcpkg then complains that the SHA512 doesn't match and tells you the SHA512 you should use. You cut and paste that into the portfile.cmake and everything starts working.

Which of course completely defeats the point of the SHA512, but we are going to ignore that.